distributed/scheduler.py

```diff
@@ -450,12 +453,17 @@ def start(self, addr_or_port=8786, start_queues=True):
                 # as it would prevent connecting via 127.0.0.1.
                 self.listen(('', addr_or_port), listen_args=self.listen_args)
                 self.ip = get_ip()
+                listen_host = ''
             else:
                 self.listen(addr_or_port, listen_args=self.listen_args)
                 self.ip = get_address_host(self.listen_address)
+                listen_host = self.ip
+
+            if self.ip == '0.0.0.0':
```

Comment:
Can `self.ip` *ever* be "0.0.0.0"? AFAIR it should be an actual IP address, not a wildcard address...